perfkitbenchmarker/linux_packages/maven.py

```diff
@@ -0,0 +1,79 @@
+import posixpath
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker.linux_packages import INSTALL_DIR
+
+try:
+  import urlparse
+except ImportError:
+  from urllib.parse import urlparse
+
+flags.DEFINE_string('maven_version', '3.6.3', 'The version of maven')
+FLAGS = flags.FLAGS
+MVN_URL = 'https://archive.apache.org/dist/maven/maven-{0}/{1}/binaries/apache-maven-{1}-bin.tar.gz'
+MVN_DIR = posixpath.join(INSTALL_DIR, "maven")
+MVN_ENV_PATH = "/etc/profile.d/maven.sh"
+
+MVN_ENV = """
+export JAVA_HOME={java_home}
+export M2_HOME={maven_home}
+export MAVEN_HOME={maven_home}
+export PATH={maven_home}/bin:$PATH
+"""
+
+
+def GetRunCommand(arguments):
+  """ Return Maven run command including proxy settings """
+  command = "source {} && mvn {}".format(MVN_ENV_PATH, arguments)
+  proxy_settings = ""
+  http_proxy_params = " -Dhttp.proxyHost={host} -Dhttp.proxyPort={port} "
+  https_proxy_params = " -Dhttps.proxyHost={host} -Dhttps.proxyPort={port} "
+
+  if FLAGS["http_proxy"].present:
+    parsed_url = urlparse.urlparse(FLAGS.http_proxy)
+    proxy_settings += http_proxy_params.format(host=parsed_url.hostname, port=parsed_url.port)
+
+  if FLAGS["https_proxy"].present:
+    parsed_url = urlparse.urlparse(FLAGS.https_proxy)
+    proxy_settings += https_proxy_params.format(host=parsed_url.hostname, port=parsed_url.port)
+
+  if proxy_settings:
+    command += proxy_settings
+  return command
+
+
+def AptInstall(vm):
+  _Install(vm)
+
+
+def YumInstall(vm):
+  vm.InstallPackages('which')
+  _Install(vm)
+
+
+def _Install(vm):
```

Comment:
Suggest adding a test for this function inside `perfkitbenchmarker/tests/linux_packages/maven_test.py`.

Reply:
Done.